src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/appointments/AppointmentsToNomisIntTest.kt

```diff
@@ -453,8 +457,344 @@ class AppointmentsToNomisIntTest : SqsIntegrationTestBase() {
+            verify(telemetryClient).trackEvent(eq("appointment-update-failed"), any(), isNull())
+            verify(telemetryClient).trackEvent(eq("appointment-update-success"), any(), isNull())
+          }
+        }
+      }
+
+      @Nested
+      inner class WhenServiceKeepsFailing {
+
+        @BeforeEach
+        fun setUp() {
+          appointmentsApi.stubGetAppointmentInstance(id = APPOINTMENT_INSTANCE_ID, response = appointmentResponse)
+          mappingServer.stubGetMappingGivenAppointmentInstanceId(APPOINTMENT_INSTANCE_ID, mappingResponse)
+          nomisApi.stubAppointmentUpdateWithError(EVENT_ID, 503)
+          publishAppointmentEvent("appointments.appointment-instance.updated")
+        }
+
+        @Test
+        fun `will callback back to appointment service 3 times before given up`() {
+          await untilAsserted {
+            appointmentsApi.verify(3, getRequestedFor(urlEqualTo("/appointment-instances/$APPOINTMENT_INSTANCE_ID")))
+          }
+        }
+
+        @Test
+        fun `will create failure telemetry`() {
+          await untilAsserted {
+            verify(telemetryClient, times(3)).trackEvent(
+              eq("appointment-update-failed"),
+              check {
+                assertThat(it["appointmentInstanceId"]).isEqualTo(APPOINTMENT_INSTANCE_ID.toString())
+                assertThat(it["nomisEventId"]).isEqualTo(EVENT_ID.toString())
+              },
+              isNull(),
+            )
+          }
+        }
+
+        @Test
+        fun `will add message to dead letter queue`() {
+          await untilCallTo {
+            awsSqsAppointmentDlqClient!!.countAllMessagesOnQueue(appointmentDlqUrl!!).get()
+          } matches { it == 1 }
+        }
+      }
+    }
+  }
+
+  @Nested
+  inner class CancelAppointment {
+    @Nested
+    inner class WhenAppointmentHasJustBeenCancelledByAppointmentService {
+
+      @BeforeEach
+      fun setUp() {
+        mappingServer.stubGetMappingGivenAppointmentInstanceId(APPOINTMENT_INSTANCE_ID, mappingResponse)
+        nomisApi.stubAppointmentCancel(EVENT_ID)
+        publishAppointmentEvent("appointments.appointment-instance.cancelled")
+      }
+
+      @Test
+      fun `will cancel an appointment in NOMIS`() {
+        await untilAsserted {
+          nomisApi.verify(putRequestedFor(urlEqualTo("/appointments/$EVENT_ID/cancel")))
+        }
+      }
+
+      @Test
+      fun `will create success telemetry`() {
+        await untilAsserted {
+          verify(telemetryClient).trackEvent(
+            eq("appointment-cancel-success"),
+            check {
+              assertThat(it["appointmentInstanceId"]).isEqualTo(APPOINTMENT_INSTANCE_ID.toString())
+              assertThat(it["nomisEventId"]).isEqualTo(EVENT_ID.toString())
+            },
+            isNull(),
+          )
+        }
+      }
+    }
+
+    @Nested
+    inner class Exceptions {
+
+      @Nested
+      inner class WhenServiceFailsOnce {
```

Comment:
I'm not sure of the benefit of the fails once / fails forever tests on every single message - aren't we just testing that DLQ retries work? What do you think @andymarke?

Reply:
I did wonder

Reply:
The original was there to test a specific scenario with SQS, also agreed we don't need it all over the place